src/widgets/modals/FinaliseModal.js

```diff
@@ -30,6 +33,27 @@ export function FinaliseModal(props) {
   const { record, recordType, checkForError, finaliseText } = props.finaliseItem;
   if (!record || !record.isValid()) return null; // Record may have been deleted
   const errorText = !record.isFinalised && checkForError && checkForError(record);
+
+  // Wrapped in try-catch block so that finalise methods in schema can throw an error
+  // as last line of defence
+  const tryFinalise = () => {
+    props.runWithLoadingIndicator(() => {
+      try {
+        if (record) {
+          props.database.write(() => {
+            record.finalise(props.database, props.user);
```

Comment:
I think just before calling record.finalise() we should check this.isFinalise and again check checkForError(record). This way we can remove all isFinalised checks in stocktake, transact etc.. and remove the extra this.itemsBelowMinimum.length > 0 check that's you've added in this PR.  I've added some extra comments to https://github.com/sussol/mobile/issues/629, should describe why.

Reply:
I like this idea but I also like keeping the checks in the finalise methods so that juuuust in case someone sometime adds a call to it outside of the finalise button, it's still safe. It's negligible overhead wise checking it twice. Though redundant. But redundancy is kind of the point I'm going for.

Reply:
Ok sure, then we should do the full pre finalise check in record.finalise right?